orttraining/orttraining/python/ort_trainer.py

```diff
@@ -773,7 +773,13 @@ def state_dict(self):
             if n.name not in torch_state:
                 torch_state[n.name] = torch.from_numpy(numpy_helper.to_array(n))
 
-        return torch_state
+        # Need to remove redundant initializers and name suffices to map back to original torch state names
+        torch_state_to_return = {}
+        for name, value in torch_state.items():
+            if not (("Moment" in name) or ("Update_Count" in name)):
+                name = name.replace('_fp16', '')
```

Comment:
Are we saving fp16 version of the model weights in pytorch state_dict? 
I am wondering why it's needed?

Reply:
BTW, not a big fan of implicit name matching. 
Can you explore if we can we this explicit mapping?

Reply:
Yea, ORT saves both fp16 weights and fp32 weights. I got a bit confused earlier. But I decided to save a copy of original torch state keys and just grab the value from ort state using exact key match..